### PR TITLE
[ADD] delivery_price_rule_untaxed

### PR DESCRIPTION
**Pull request by AshishHirapara**
migration to 14.0

### CODE REVIEW

File: delivery_price_rule_untaxed/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2018 Simone Rubino - Agile Business Group
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Delivery price rule untaxed",
+    "summary": "Add untaxed amount to variables for price delivery price rule",
+    "version": "14.0.1.0.0",
+    "development_status": "Beta",
+    "category": "Stock",
+    "website": "https://github.com/OCA/delivery-carrier"
```
Comment (rousseldenis) [inline delivery_price_rule_untaxed/__manifest__.py:8]:
```suggestion
    "website": "https://github.com/OCA/delivery-carrier/tree/"
```

File: delivery_price_rule_untaxed/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Copyright 2018 Simone Rubino - Agile Business Group
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+{
+    "name": "Delivery price rule untaxed",
+    "summary": "Add untaxed amount to variables for price delivery price rule",
+    "version": "14.0.1.0.0",
+    "development_status": "Beta",
```
Comment (rousseldenis) [inline delivery_price_rule_untaxed/__manifest__.py]: this is the default, so not needed

File: delivery_price_rule_untaxed/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    "category": "Stock",
+    "website": "https://github.com/OCA/delivery-carrier"
+    "14.0/delivery_price_rule_untaxed",
+    "author": "Agile Business Group, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "application": False,
+    "installable": True,
+    "depends": ["delivery"],
+    "maintainers": ["AshishHirapara"],
+    "contributors": ["Ashish Hirpara"],
```
Comment (rousseldenis) [inline delivery_price_rule_untaxed/__manifest__.py]: Remove this line, CONTRIBUTORS.rst is sufficent

File: delivery_price_rule_untaxed/data/delivery_price_rule_untaxed_demo.xml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <data noupdate="1">
```
Comment (rousseldenis) [inline delivery_price_rule_untaxed/data/delivery_price_rule_untaxed_demo.xml]: You can just use one line:
```suggestion
    <odoo noupdate="1">
```

File: delivery_price_rule_untaxed/tests/test_delivery_price_rule_untaxed.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+                (
+                    "product_id",
+                    "=",
+                    self.sale_untaxed_delivery_charges.carrier_id.product_id.id,
+                ),
+            ]
+        )
+
+        self.assertEqual(len(line), 1, "Delivery cost is not Added")
+        # self.assertEqual(
```
Comment (rousseldenis) [inline delivery_price_rule_untaxed/tests/test_delivery_price_rule_untaxed.py]: Please remove this if not used

**Review by rousseldenis — CHANGES_REQUESTED** [review]
LGTM. Some minor changes.

File: delivery_price_rule_untaxed/models/delivery_carrier.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = "delivery.carrier"
+
+    def _get_price_available(self, order):
+        self.ensure_one()
+        self = self.with_context({"order_amount_untaxed": order.amount_untaxed})
+        res = super(DeliveryCarrier, self)._get_price_available(order)
+        return res
```
Comment (MeetKD) [inline delivery_price_rule_untaxed/models/delivery_carrier.py]:
```suggestion
        return super(DeliveryCarrier, **self)._get_price_available(order)
```

Comment (rousseldenis) [reply delivery_price_rule_untaxed/models/delivery_carrier.py]: I didn't see that one.

But no, the good one is :
```suggestion
        new_self = self.with_context({"order_amount_untaxed": order.amount_untaxed})
        return super(DeliveryCarrier, new_self)._get_price_available(order)
```
As you cannot replace self.

File: delivery_price_rule_untaxed/__manifest__.py (hunk 1 above)
Comment (AshishHirapara) [reply delivery_price_rule_untaxed/__manifest__.py:8]: I have tried to update the website, but hook id: oca-fix-manifest-website is removing the '/tree/' from the URL. Suggest the solution or alternative way.

Comment (rousseldenis) [reply delivery_price_rule_untaxed/__manifest__.py:8]: Just change to https://github.com/OCA/delivery-carrier

File: delivery_price_rule_untaxed/models/delivery_carrier.py (hunk 6 above)
Comment (AshishHirapara) [reply delivery_price_rule_untaxed/models/delivery_carrier.py]: done, thanks

File: delivery_price_rule_untaxed/__manifest__.py (hunk 3 above)
Comment (AshishHirapara) [reply delivery_price_rule_untaxed/__manifest__.py]: done

File: delivery_price_rule_untaxed/__manifest__.py (hunk 2 above)
Comment (AshishHirapara) [reply delivery_price_rule_untaxed/__manifest__.py]: done

File: delivery_price_rule_untaxed/data/delivery_price_rule_untaxed_demo.xml (hunk 4 above)
Comment (AshishHirapara) [reply delivery_price_rule_untaxed/data/delivery_price_rule_untaxed_demo.xml]: done

File: delivery_price_rule_untaxed/tests/test_delivery_price_rule_untaxed.py (hunk 5 above)
Comment (AshishHirapara) [reply delivery_price_rule_untaxed/tests/test_delivery_price_rule_untaxed.py]: done

**Review by francesco-ooops — APPROVED** [review]
LGTM!

**Review by elvise — APPROVED** [review]
LGTM!

**Comment by francesco-ooops** [comment]
@rousseldenis merge?

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**